test/CirclesBackingFactory.t.sol

```diff
@@ -3,65 +3,557 @@ pragma solidity ^0.8.28;
+        // Attempt to create LBP => revert
+        vm.expectRevert(CirclesBacking.InsufficientBackingAssetBalance.selector);
+        _createLBP(predictedInstance);
+    }
+
+    function test_ReleaseBalancerPoolTokens() public {
+        address predictedInstance = _initUserWithBackedCRC(TEST_ACCOUNT, WETH);
+
+        // Simulate fill
+        _simulateCowSwapFill(predictedInstance, WETH, WETH_DEAL_AMOUNT);
+
+        // Create LBP
+        _createLBP(predictedInstance);
+
+        // Warp enough time so that release is possible
+        vm.warp(block.timestamp + YEAR);
+
+        address lbp = CirclesBacking(predictedInstance).lbp();
+        uint256 frozenLPTokensAmount = IERC20(lbp).balanceOf(predictedInstance);
+
+        assertTrue(factory.isActiveLBP(TEST_ACCOUNT));
+        // Release from backer
+        vm.prank(TEST_ACCOUNT);
+        CirclesBacking(predictedInstance).releaseBalancerPoolTokens(TEST_ACCOUNT);
+
+        assertFalse(factory.isActiveLBP(TEST_ACCOUNT));
+        assertEq(IERC20(lbp).balanceOf(predictedInstance), 0);
+        assertEq(IERC20(lbp).balanceOf(TEST_ACCOUNT), frozenLPTokensAmount);
+    }
+
+    function test_RevertIf_ReleaseIsCalledNotByBackingContract() public {
+        address predictedInstance = _initUserWithBackedCRC(TEST_ACCOUNT, WETH);
+
+        // Simulate fill
+        _simulateCowSwapFill(predictedInstance, WETH, WETH_DEAL_AMOUNT);
+
+        // Create LBP
+        _createLBP(predictedInstance);
+
+        vm.expectRevert(CirclesBackingFactory.OnlyCirclesBacking.selector);
+        factory.notifyRelease(predictedInstance);
+    }
+
+    function test_RevertIf_ReleaseBalancerPoolDeadlineNotMet() public {
+        address predictedInstance = _initUserWithBackedCRC(TEST_ACCOUNT, WETH);
+
+        // Simulate fill
+        _simulateCowSwapFill(predictedInstance, WETH, WETH_DEAL_AMOUNT);
+
+        // Create LBP
+        _createLBP(predictedInstance);
+
+        // Attempt to release too soon
+        vm.prank(TEST_ACCOUNT);
+        vm.expectRevert(
+            abi.encodeWithSelector(CirclesBacking.TokensLockedUntilTimestamp.selector, block.timestamp + YEAR)
+        );
+        CirclesBacking(predictedInstance).releaseBalancerPoolTokens(TEST_ACCOUNT);
+    }
+
+    function test_RevertIf_NotBacker() public {
+        address predictedInstance = _initUserWithBackedCRC(TEST_ACCOUNT, WETH);
+
+        // Simulate fill
+        _simulateCowSwapFill(predictedInstance, WETH, WETH_DEAL_AMOUNT);
+
+        // Create LBP
+        _createLBP(predictedInstance);
+        vm.warp(block.timestamp + YEAR);
+
+        // Some random address tries to release
+        vm.prank(address(0));
+        vm.expectRevert(CirclesBacking.NotBacker.selector);
+        CirclesBacking(predictedInstance).releaseBalancerPoolTokens(TEST_ACCOUNT);
+    }
+
+    function test_RevertIf_BackingAssetIsNotSupported() public {
```

Comment:
i'm a little bit confuse with the test name, the test tries to call factory function, which is allowed to be called only by factory deployed instances.